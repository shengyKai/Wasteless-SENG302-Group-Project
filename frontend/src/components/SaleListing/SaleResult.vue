<template>
  <v-list-item>
    <v-list-item-avatar>
      <p>img</p>
    </v-list-item-avatar>
    <v-list-item-content>
      <v-list-item-title>
        <label class="result-title">The Nathan Apple</label>
        <label class="result-title-join">From </label>
        <label class="result-title-business">Nathan Apple LTD</label>
      </v-list-item-title>
      <v-list-item-subtitle>
        <label>More Info: This is a Nathan Apple. The absolute rarest of its kind. You will not find a Nathan Apple anywhere else.</label>
      </v-list-item-subtitle>
      <v-list-item-subtitle>
        <label class="total-price-label">Total Price:</label>
        <label>$10.09</label>
        <label class="divider1"/>
        <label class="quantity-label">Quantity:</label>
        <label>109</label>
      </v-list-item-subtitle>
      <v-list-item-subtitle>
        <label class="creation-date-label">Creation Date:</label>
        <label>10 Jan 2021</label>
        <label class="divider2"/>
        <label class="closing-date-label">Closing Date:</label>
        <label>15 Jan 2021</label>
      </v-list-item-subtitle>
    </v-list-item-content>
  </v-list-item>

  <!-- <v-card>
    <div class="sale-image">
      <p>img</p>
    </div>
    <div class="title">
      <v-card-title class="saleTitle">The Nathan Apple</v-card-title>
      <label class="saleTitleJoin"> From </label>
      <a class="businessLink">Nathan Apple LTD</a>
    </div>
    <div class="fields">
      <v-row no-gutters>
        <v-col cols="1">
          <label>Total Price:</label>
        </v-col>
        <v-col cols="1">
          <label>$10.09</label>
        </v-col>
        <div class="vertical-line"/>
        <v-col cols="1">
          <label>Date Created:</label>
        </v-col>
        <v-col cols="1">
          <label>10 Jan 2021</label>
        </v-col>
        <div class="vertical-line"/>

      </v-row>


      <v-row no-gutters>
        <v-col class="column" cols="4">
          <label class="field-label">More Info:</label>
        </v-col>
        <v-col class="column" cols="8">
          <label class="field-element">This is a Nathan Apple. The absolute rarest of its kind. You will not find a Nathan Apple anywhere else.</label>
        </v-col>
        <v-col class="column" cols="4">
          <label class="field-label">Total Price:</label>
        </v-col>
        <v-col class="column" cols="8">
          <label class="field-element">$10.09</label>
        </v-col>
        <v-col class="column" cols="4">
          <label class="field-label">Quantity:</label>
        </v-col>
        <v-col class="column" cols="8">
          <label class="field-element">109</label>
        </v-col>
        <v-col class="column" cols="4">
          <label class="field-label">Date Created:</label>
        </v-col>
        <v-col class="column" cols="8">
          <label class="field-element">9 Jan 2021</label>
        </v-col>
        <v-col class="column" cols="4">
          <label class="field-label">Closing Date:</label>
        </v-col>
        <v-col class="column" cols="8">
          <label class="field-element">10 Jan 2021</label>
        </v-col>
        <v-col class="column" cols="4">
          <label class="field-label">Expiry Date:</label>
        </v-col>
        <v-col class="column" cols="8">
          <label class="field-element">11 Jan 2021</label>
        </v-col>
      </v-row>
  </div>
  </v-card> -->
</template>

<script>
export default {
  name: "SaleResult",
  data() {
    return {
    };
  }
};
</script>

<style scoped>
.result-title {
  font-size: 19px;
  font-weight: bold;
}

.result-title-join {
  font-size: 15px;
  margin-left: 10px;
}

.result-title-business {
  font-size: 15px;
}

.divider1 {
  margin-right: 64px;
}

.divider2 {
  margin-right: 30px;
}

.total-price-label {
  margin-right: 28px;
}

.quantity-label {
  margin-right: 37px;
}

.creation-date-label {
  margin-right: 10px;
}

.closing-date-label {
  margin-right: 10px;
}

.vertical-line {
  border-left: 1px solid black;
  margin-top: 3px;
  margin-right: 10px;
  height: 15px;
}
</style>