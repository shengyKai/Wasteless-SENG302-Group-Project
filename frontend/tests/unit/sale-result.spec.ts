import Vue from 'vue';
import Vuetify from 'vuetify';
import { createLocalVue, mount, Wrapper } from '@vue/test-utils';
import SaleResult from "@/components/SaleListing/SaleResult.vue";

Vue.use(Vuetify);

jest.mock('@/api/currency', () => ({
  currencyFromCountry: jest.fn(() => {
    return {
      code: 'Currency code',
      symbol: 'Currency symbol'
    };
  })
}));

describe('SaleResult.vue', () => {
  let wrapper: Wrapper<any>;
  let vuetify: Vuetify;

  beforeEach(async () => {
    jest.clearAllMocks();
    const localVue = createLocalVue();
    vuetify = new Vuetify();

    wrapper = mount(SaleResult, {
      localVue,
      vuetify,
      stubs: ['FullSaleListing', 'router-link'],
      propsData: {
        saleItem: {
          "id": 57,
          "inventoryItem": {
            "id": 101,
            "product": {
              "business": {
                "id": 5,
                "name": "This Inc.",
              },
              "id": "WATT-420-BEANS",
              "name": "Watties Baked Beans - 420g can",
              "description": "Baked Beans as they should be.",
              "manufacturer": "Heinz Wattie's Limited",
              "recommendedRetailPrice": 2.2,
              "created": "2021-05-15T05:55:32.808Z",
              "countryOfSale": "New Zealand",
              "images": [],
            },
            "quantity": 4,
            "pricePerItem": 6.5,
            "totalPrice": 21.99,
            "manufactured": "2021-05-15",
            "sellBy": "2021-05-15",
            "bestBefore": "2021-05-15",
            "expires": "2021-05-15"
          },
          "quantity": 3,
          "price": 17.99,
          "moreInfo": "Seller may be willing to consider near offers.",
          "created": "2021-07-14T11:44:00Z",
          "closes": "2021-07-21T23:59:00Z",
          "interestedCount": "5"
        },
        businessId: 1
      }
    });
  });

  it('FullSaleListing should not initial be shown', async () => {
    expect(wrapper.findComponent({name: 'FullSaleListing'}).exists()).toBeFalsy();
  });

  describe('Title clicked', () => {
    beforeEach(async () => {
      const title = wrapper.findComponent({ref: 'title'});
      await title.trigger('click');
      await Vue.nextTick();

    });

    it('FullSaleListing should be shown after title is clicked', () => {
      expect(wrapper.findComponent({name: 'FullSaleListing'}).exists()).toBeTruthy();
    });

    it('When "goBack" is emitted by FullSaleListing the FullSaleListing should be hidden', async () => {
      wrapper.findComponent({name: 'FullSaleListing'}).vm.$emit('goBack');
      await Vue.nextTick();

      expect(wrapper.findComponent({name: 'FullSaleListing'}).exists()).toBeFalsy();
    });

    it('When "refresh" is emitted by FullSaleListing the event should be passed on', async () => {
      expect(wrapper.emitted().refresh).toBeFalsy();

      wrapper.findComponent({name: 'FullSaleListing'}).vm.$emit('refresh');
      await Vue.nextTick();

      expect(wrapper.emitted().refresh).toBeTruthy();
    });
  });

  it('Component matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});