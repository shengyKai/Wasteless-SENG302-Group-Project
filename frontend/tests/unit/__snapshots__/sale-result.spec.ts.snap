// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaleResult.vue Component matches snapshot 1`] = `
<div businessid="1">
  <div tabindex="-1" class="v-list-item theme--light purple--text">
<<<<<<< HEAD
    <div class="v-avatar v-list-item__avatar" style="height: 70px; min-width: 40px; width: 70px;"><img src="../../assets/grumpy.webp"></div>
    <div class="v-list-item__content">
      <div class="v-list-item__title"><a class="result-title">Watties Baked Beans - 420g can</a> <label class="result-title-join">From </label> <label class="result-title-business">test_business</label></div>
      <div class="v-list-item__subtitle"><label>More Info: Seller may be willing to consider near offers.</label></div>
      <div class="v-list-item__subtitle"><label class="total-price-label">Total Price:</label> <label>$17.99</label> <label class="divider1"></label> <label class="quantity-label">Quantity:</label> <label>3</label></div>
      <div class="v-list-item__subtitle"><label class="creation-date-label">Creation Date:</label> <label>14 Jul 2021</label> <label class="divider2"></label> <label class="closing-date-label">Closing Date:</label> <label>22 Jul 2021</label></div>
=======
    <div class="pl-0 col col-auto align-self-center">
      <div class="v-avatar white--text headline secondary" style="height: 80px; min-width: 80px; width: 80px; font-size: 40px;">W</div>
    </div>
    <div class="col">
      <div class="row row--dense">
        <div class="col">
          <h4><a>Watties Baked Beans - 420g can</a>
            from
            <router-link-stub to="/business/5" class="text--secondary">
              This Inc.
            </router-link-stub>
          </h4>
        </div>
      </div>
      <div class="row my-0 py-0">
        <div class="my-0 py-0 col"><label class="font-weight-bold">More Info:</label> Seller may be willing to consider near offers.
        </div>
      </div>
      <div class="row mt-0 row--dense">
        <div class="font-weight-bold col col-auto">
          Total Price:
          <br>
          Creation Date:
        </div>
        <div class="col col-auto">
          $17.99
          <br>
          14 Jul 2021
        </div>
        <div class="font-weight-bold col col-auto">
          Quantity:
          <br>
          Closing Date:
        </div>
        <div class="col col-auto">
          3
          <br>
          22 Jul 2021
        </div>
      </div>
>>>>>>> dev
    </div>
  </div>
</div>
`;
